src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -257,16 +249,16 @@ default MethodHandle downcallHandle(Addressable symbol, FunctionDescriptor funct
      *
      * @param target the target method handle.
      * @param function the upcall stub function descriptor.
-     * @param session the upcall stub memory session.
-     * @return a zero-length segment whose base address is the address of the upcall stub.
+     * @param scope the scope associated with the returned upcall stub segment.
+     * @return a zero-length segment whose address is the address of the upcall stub.
      * @throws IllegalArgumentException if the provided function descriptor is not supported by this linker.
      * @throws IllegalArgumentException if it is determined that the target method handle can throw an exception, or if the target method handle
      * has a type that does not match the upcall stub <a href="Linker.html#upcall-stubs"><em>inferred type</em></a>.
-     * @throws IllegalStateException if {@code session} is not {@linkplain MemorySession#isAlive() alive}.
-     * @throws WrongThreadException if this method is called from a thread other than the thread
-     * {@linkplain MemorySession#ownerThread() owning} {@code session}.
+     * @throws IllegalStateException if {@code scope} is not {@linkplain SegmentScope#isAlive() alive}.
+     * @throws WrongThreadException if this method is called from a thread {@code T},
+     * such that {@code scope.isAccessibleBy(T) == false}.
      */
-    MemorySegment upcallStub(MethodHandle target, FunctionDescriptor function, MemorySession session);
+    MemorySegment upcallStub(MethodHandle target, FunctionDescriptor function, SegmentScope scope);
 
     /**
      * Returns a symbol lookup for symbols in a set of commonly used libraries.
```

Comment:
Use {@return ...}

Reply:
Same - I'll leave these tweaks for later.